### ISSUE TITLE
Escape comma inside a column

### ISSUE THREAD

**Issue by jkevinturado** [adaltas/node-csv#346]
Hi,

My raw data got a comma from a certain field, While I'm parsing the data it cause me to  have a new column instead. How do you escape a comma inside a column to prevent this?

Here is a  example of raw csv data

`"test@email.org, test2@mail.com",email.org,none,bad
`
now the data returned from me is

`[test@email.org,test2@mail.com,email.org,none,bad]`

What I'm expecting is 

`["test@email.org,test2@mail.com",email.org,none,bad]`

I tried to escape the comma by `escape: '\\,'` but it doesn't help

**Comment by jkevinturado** [comment]
Yes that is correct, is this a default config?

**Comment by jkevinturado** [comment]
Hi,

I am escaping the double quotes and I need to escape that because some of the fields do have double quotes inside the field for example: 

![image](https://user-images.githubusercontent.com/42071552/174522298-806715de-2100-4309-8976-6a9ec12b9e7f.png)

So this is my config btw:

![image](https://user-images.githubusercontent.com/42071552/174522385-f7206a16-2ae6-424a-8a59-63b9ec41e135.png)

**Comment by jkevinturado** [comment]
Hi, I'm using the default config.

Now the error is:
`Invalid Closing Quote: got "d" at line 1 instead of delimiter, record delimiter, trimable character (if activated) or comment`

Because I have this double quote inside a field:
![image](https://user-images.githubusercontent.com/42071552/174527981-3e54623a-5e97-4e18-b102-ea73352e6daa.png)

**Comment by wdavidw** [comment]
The screenshot I shared use the default config.

**Comment by wdavidw** [comment]
Please share a minimalist and complete example for additionnal help.

**Comment by jkevinturado** [comment]
Hi, 

 I was using the default config, I'm okay with the first issue which has the comma inside, now I have double quotes inside the field and it says
![image](https://user-images.githubusercontent.com/42071552/174563280-3925bee1-7060-4c93-8532-ef76d245f225.png)

Here is the input please focus on the 2nd line which has the double quotes inside the field

![image](https://user-images.githubusercontent.com/42071552/174563405-0e665760-3717-411f-b7fd-6e962d88ee9e.png)

**Comment by wdavidw** [comment]
Honestly, I don't think there is much we can do to handle such a scenario. The first quote place the parser in quoting mode and second quote is not escape which makes the field invalid

**Comment by jkevinturado** [comment]
I can actually escape the comma, using the option `escape`. Can I escape multiple characters?

I've tried `escape: ',"'` but I didn't work

**Comment by wdavidw** [comment]
Yes, escape may be composed of multiple characters: https://github.com/adaltas/node-csv/blob/master/packages/csv-parse/test/option.escape.coffee#L69-L79

**Comment by jkevinturado** [comment]
how?

**Comment by wdavidw** [comment]
Look at the tests with the link above, I based my answer on them.

**Comment by jkevinturado** [comment]
it didn't work, I think the link above you've provided will work if the character is contiguous. 

Mine have to escape comma and double quotes just like this `escape: ',"'` unfortunately this doesn't work also as well

**Comment by patelkhushal18** [comment]
For me, I had to escape both `,` and `\n`. I ended up removing commas and \n inside matching double quotes before parsing:
```
// remove commas and newlines inside double-quoted fields
const cleanedData = data.replace(/"([^"]*?)"/g, match => match.replace(/[,\n]/g, ''));

const records = csvParse.parse(cleanedData, {
  columns: true,      // To treat first row as headers
  relax_quotes: true, // Allow quotes within quotes
});
```
